### PR TITLE
feat(csv): add installmodes to CSV spec

### PR DESCRIPTION
**Pull request by njhale**
- Adds InstallModes to CSV spec.
- Adds InstallModes to all current operators.

Addresses [ALM-842](https://jira.coreos.com/browse/ALM-842) and [ALM-843](https://jira.coreos.com/browse/ALM-843)

/assign @alecmerdler @ecordell

### CODE REVIEW

**Comment by ecordell** [comment]
Do you need to run codegen on this?

**Comment by njhale** [comment]
> Do you need to run codegen on this?

Yes, I do. I was planning on running it after I finish writing tests.

File: pkg/controller/operators/olm/operatorgroup.go [hunk 1]
```diff
@@ -57,6 +58,17 @@ func (a *Operator) syncOperatorGroups(obj interface{}) error {
...
 
 	for _, csv := range a.csvSet(op.Namespace, v1alpha1.CSVPhaseSucceeded) {
+		// Check if the CSV's InstallModes support the operator group
+		modeSet, err := v1alpha1.NewInstallModeSet(csv.Spec.InstallModes)
+		if err != nil {
+			a.Log.Errorf("CSV has invalid InstallModes: %v", err)
+			continue
+		}
+		if err := modeSet.Supports(targetNamespaces); err != nil {
+			a.Log.Debugf("CSV InstallModes do not support operator group: %v", err)
```
[the remark this answers is not in the document]
Comment (njhale) [reply pkg/controller/operators/olm/operatorgroup.go]: that's a good idea.

**Pull request merged**